flowmachine/flowmachine/core/model_result.py

```diff
@@ -150,25 +152,50 @@ def to_sql(
 
         def do_query() -> ModelResult:
             logger.debug("Getting storage lock.")
-            with rlock(self.redis, self.md5):
+            q_state_machine = QueryStateMachine(self.redis, self.md5)
+            current_state, this_thread_is_owner = q_state_machine.execute()
+            if this_thread_is_owner:
                 logger.debug("Obtained storage lock.")
```

Comment:
Does this line about the storage lock actually still apply?

Reply:
Good point. Will change.